File: Data Structures & Algorithms/trapping-rain-water/submission-1.py
class Solution:
    def trap(self, height: List[int]) -> int:
        myStack = []

        area = 0

        for i in range(len(height)):
            while myStack and height[i] >= height[myStack[-1]]:
                print(i, myStack)
                bottom = height[myStack.pop()]

                if myStack:
                    h = min(height[i], height[myStack[-1]]) - bottom
                    w = i - myStack[-1] - 1
                    area += h*w
            
            myStack.append(i)

        return area

File: Data Structures & Algorithms/min-cost-climbing-stairs/submission-1.java
class Solution {
    public int minCostClimbingStairs(int[] cost) {
        int n = cost.length;

        int[] dp = new int[n+1];

        for (int i = 2; i<= n; i++)
        {
            int c1 = cost[i-2] + dp[i-2];
            int c2 = cost[i-1] + dp[i-1];
            dp[i] = Math.min(c1, c2);
        }

        return dp[n];
    }
}
